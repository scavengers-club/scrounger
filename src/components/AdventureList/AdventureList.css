.list-img {
  width: 35vw;
  height: 35vw;
  object-fit: cover;
  border-radius: 15px;
  border: 2px solid var(--sand);
  filter: drop-shadow(5px 5px 5px rgb(62, 62, 62));
}

.adventure-list {
  display: flex;
  flex-wrap: wrap;
  align-items: center;
  justify-content: space-evenly;
  gap: 5px;
}

.adventure-card {
  max-width: 40vw;
  text-align: center;
}

.list-title {
  font-family: 'Reenie Beanie', cursive;
  font-size: 150%;
  text-transform: uppercase;
  margin: 5px;
  text-shadow: 2px 2px 2px rgb(62, 62, 62);
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
}

.list-link {
  text-decoration: none;
}
