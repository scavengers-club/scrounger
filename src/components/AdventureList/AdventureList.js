import React from 'react';

//this will render all of the adventures onto to home page

export default function AdventureList({ adventures }) {
  return (
    <div className="adventure-list">
      {adventures.map((adventure) => (
        <div key={adventure.id}>
          <h3>{adventure.name}</h3>
          <img src={adventure.image}></img>
        </div>
      ))}
    </div>
  );
}
