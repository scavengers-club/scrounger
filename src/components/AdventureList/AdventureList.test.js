import { render, screen } from '@testing-library/react';
import { MemoryRouter, Route } from 'react-router-dom';
import AdventureList from './AdventureList';

test('<AdventureList/> should render the list of adventures', async () => {
  const { container } = render(
    <MemoryRouter initialEntries={['/']}>
      <Route path="/">
        <AdventureList
          adventures={[
            {
              id: 1,
              name: 'unique adventure',
              description: 'this is a description',
              image:
                'https://media.istockphoto.com/photos/sunset-in-downtown-portland-oregon-picture-id1283364803?b=1&k=20&m=1283364803&s=170667a&w=0&h=CEcB5sIBGs1GX3LhyLxgU53K7rgeP4vNsC4tWhesOAo=',
              hint_1: 'go here',
              hint_2: 'go there',
              hint_3: 'go everywhere',
              solution: 'you are beautiful',
            },
          ]}
        />
      </Route>
    </MemoryRouter>
  );

  await screen.findByText('unique adventure');
  expect(container).toMatchSnapshot();
});
