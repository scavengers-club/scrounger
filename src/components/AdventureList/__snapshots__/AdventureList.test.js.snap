// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AdventureList/> should render the list of adventures 1`] = `
<div>
  <div
    class="adventure-list"
  >
    <div
      class="adventure-card"
    >
      <a
        class="list-link"
        href="/adventure/1"
      >
        <h3
          class="list-title"
        >
          unique adventure
        </h3>
        <img
          alt="unique adventure"
          class="list-img"
          src="https://media.istockphoto.com/photos/sunset-in-downtown-portland-oregon-picture-id1283364803?b=1&k=20&m=1283364803&s=170667a&w=0&h=CEcB5sIBGs1GX3LhyLxgU53K7rgeP4vNsC4tWhesOAo="
        />
      </a>
    </div>
  </div>
</div>
`;
